src/utils/general.ts

```diff
@@ -0,0 +1,8 @@
+import { replacer } from './marshaling';
+
+/**
+ * Deeply compares two objects, ignoring sorting of keys
+ * @param a the first object
+ * @param b the second object
+ */
+export const deepEquals = (a: any, b: any) => JSON.stringify(a, replacer, 2) === JSON.stringify(b, replacer, 2);
```

Comment:
I'd reccommend using this: https://lodash.com/docs/4.17.15#isEqual

Reply:
This repository does not currently have lodash as a dependency and I'm hoping to not have to add it.
The ordering-agnostic equality check wasn't really necessary, so I've removed it.

Reply:
I understand not wanting to add lodash due to some performance concerns but I don't think that we have any here. This is not user-facing and this is not deployed anywhere. This will be run from the developer's machine, right?

Reply:
argh, nevermind, I'll add back the check with lodash.

Reply:
So no, isEqual doesn't work because keys are out of order.